ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/locator/AllHostsLocator.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.physical.scalablepush.locator;
+
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import io.confluent.ksql.util.QueryMetadata;
+import java.net.URI;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.kafka.streams.state.HostInfo;
+import org.apache.kafka.streams.state.StreamsMetadata;
+
+/**
+ * This locator contacts all hosts since there currently isn't a mechanism to find which hosts own
+ * a particular persistent query task.
+ */
+public class AllHostsLocator implements PushLocator {
+
+  private final Supplier<List<PersistentQueryMetadata>> allPersistentQueries;
+  private final URL localhost;
+
+  public AllHostsLocator(final Supplier<List<PersistentQueryMetadata>> allPersistentQueries,
+      final URL localhost) {
+    this.allPersistentQueries = allPersistentQueries;
+    this.localhost = localhost;
+  }
+
+
+  public Set<KsqlNode> locate() {
+    final List<PersistentQueryMetadata> currentQueries = allPersistentQueries.get();
+    if (currentQueries.isEmpty()) {
+      return Collections.emptySet();
+    }
+
+    return currentQueries.stream()
+        .map(QueryMetadata::getAllMetadata)
+        .filter(Objects::nonNull)
+        .flatMap(Collection::stream)
+        .filter(streamsMetadata -> streamsMetadata != StreamsMetadata.NOT_AVAILABLE)
+        .map(StreamsMetadata::hostInfo)
+        .map(hi -> new Node(isLocalhost(hi), buildLocation(hi)))
+        .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
out of curiosity, why `LinkedHashSet`? do we care about ordering? (if so we should probably comment about why)